server/handlers.go

```diff
@@ -716,6 +721,27 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 		return
 	}
 
+	// Check for code challenge validity
+	if authCode.CodeChallenge != "" {
+		if verifier == "" {
+			s.tokenErrHelper(w, errAccessDenied, "Require code_verifier", http.StatusUnauthorized)
```

Comment:
According to [spec](https://tools.ietf.org/html/rfc6749#section-5.2), if client doesn't pass code_verifier it would be invalid_request instead of access denied. Let me know if i'm mistaken.